### PR TITLE
fix: typescriptify request.js

### PR DESCRIPTION
**Pull request by kyegupov**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Typescriptify request.js

### CODE REVIEW

File: src/lib/request/request.ts [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
-var config = require('../config');
-var getProxyForUrl = require('proxy-from-env').getProxyForUrl;
-var ProxyAgent = require('proxy-agent');
-var analytics = require('../analytics');
-
-function makeRequest(payload) {
-  return new Promise(function (resolve, reject) {
-    var body = payload.body;
-    var data;
+import debugModule = require('debug');
```
Comment (dkontorovskyy) [inline src/lib/request/request.ts]: I would do
`import {debug as debugModule} from 'debug'`

File: src/lib/request/request.ts [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
+import * as needle from 'needle';
+import { parse } from 'url';
+import { format } from 'url';
```
Comment (dkontorovskyy) [inline src/lib/request/request.ts]: Line 3 and 4 can be merged to 1 line

File: src/lib/request/request.ts [hunk 3]
```diff
@@ -1,30 +1,34 @@
...
+import { Agent } from 'http';
+import { Global } from '../../cli/args';
+import { Payload } from './types';
+
+const debug = debugModule('snyk:req');
+const snykDebug = debugModule('snyk');
+
+declare const global: Global;
+
+export = function makeRequest(payload: Payload) {
```
Comment (dkontorovskyy) [inline src/lib/request/request.ts]: Isn't `export=` old notation ?

File: src/lib/request/request.ts [hunk 4]
```diff
@@ -89,14 +93,14 @@ function makeRequest(payload) {
...
+    needle.request(method, url, data, options, (err, res, respBody) => {
       debug(err);
-      debug('response (%s): ', (res || {}).statusCode, JSON.stringify(body));
+      debug('response (%s): ', (res || {}).statusCode, JSON.stringify(respBody));
       if (err) {
         return reject(err);
       }
 
-      resolve({res: res, body: body});
+      resolve({res, respBody});
```
Comment (dkontorovskyy) [inline src/lib/request/request.ts]: You changed return key from `body` to `respBody`. Is it wanted behavior?

File: src/lib/request/request.ts (hunk 3 above)
Comment (kyegupov) [reply src/lib/request/request.ts]: yes. But we cannot change it without changing all the clients. I'd rather first typescriptify everything then convert them all.

File: src/lib/request/request.ts (hunk 4 above)
Comment (kyegupov) [reply src/lib/request/request.ts]: Nope. Fixed.

File: src/lib/request/request.ts (hunk 1 above)
Comment (kyegupov) [reply src/lib/request/request.ts]: Great idea!

**Comment by JonasFaure** [comment]
Hi :wave:

Sorry to just drop here w/o reasons really  🤷‍♂ 
I'm curious, is there a reason why `@types/agent-base` was added as a runtime dependency, and not a dev one ? @kyegupov @dkontorovskyy

**Pull request merged**